File: javascript/cluster_management/src/create-multi-region.js
import { DSQLClient } from "@aws-sdk/client-dsql"; 
import { CreateMultiRegionClustersCommand } from "@aws-sdk/client-dsql"; 

async function createMultiRegionCluster(client, linkedRegionList, witnessRegion, clusterProperties) {
    const createMultiRegionClustersCommand = new CreateMultiRegionClustersCommand({
        linkedRegionList: linkedRegionList,
        witnessRegion: witnessRegion,
        clusterProperties: clusterProperties
    });
    try {
        const response = await client.send(createMultiRegionClustersCommand);
        return response;
    } catch (error) {
        console.error("Failed to create multi-region cluster: ", error.message);
    }
}

async function main() {
    const region = "us-east-1";
    const client = new DSQLClient({
        region
    });
    const linkedRegionList = ["us-east-1", "us-east-2"];
    const witnessRegion = "us-west-2";
    const clusterProperties = {
        "us-east-1": { tags: { "Name": "Foo" }, deletionProtectionEnabled: true },
        "us-east-2": { tags: { "Name": "Bar" }, deletionProtectionEnabled: true }
    };

    const response = await createMultiRegionCluster(client, linkedRegionList, witnessRegion, clusterProperties);
    console.log("Linked Cluster ARNs:", response.linkedClusterArns);
}

main();

File: ruby/cluster_management/lib/create_multi_region.rb
require 'aws-sdk-core'
require 'aws-sdk-dsql'

def create_multi_region_cluster(region)
  us_east_1_props = {
    deletion_protection_enabled: false,
    tags: {
      'Name' => 'use1-example-cluster',
      'Usecase' => 'testing-mr-use1'
    }
  }
  
  us_east_2_props = {
    deletion_protection_enabled: false,
    tags: {
      'Name' => 'use2-example-cluster',
      'Usecase' => 'testing-mr-use2'
    }
  }
  
  begin
    # Create client with default configuration and credentials
    client = Aws::DSQL::Client.new(region: region)
    response = client.create_multi_region_clusters(
      linked_region_list: ['us-east-1', 'us-east-2'],
      witness_region: 'us-west-2',
      cluster_properties: {
        'us-east-1' => us_east_1_props,
        'us-east-2' => us_east_2_props
      }
    )
    
    # Extract cluster ARNs from the response
    arns = response.linked_cluster_arns
    raise "Expected 2 cluster ARNs, got #{arns.length}" unless arns.length == 2
        
    arns
  rescue Aws::Errors::ServiceError => e
    raise "Failed to create multi-region clusters: #{e.message}"
  end
end
